init docs for `HTMLElement.translate` (#31442)
mdn/content: files/en-us/web/api/htmlelement/translate/index.md

---
title: "HTMLElement: translate property"
short-title: translate
slug: Web/API/HTMLElement/translate
page-type: web-api-instance-property
browser-compat: api.HTMLElement.translate
---

{{APIRef("HTML DOM")}}

> The **`translate`** property of the {{domxref("HTMLElement")}} interface represents a boolean value indicating whether an element's attribute values and the values of its Text node children are to be translated when the page is localized, or whether to leave them unchanged.

Reviewer: ```suggestion
The **`translate`** property of the {{domxref("HTMLElement")}} interface indicates whether an element's attribute values and the values of its {{domxref("Text")}} node children are to be translated when the page is localized, or whether to leave them unchanged.
```

It reflects the value of the [`translate`](/en-US/docs/Web/HTML/Global_attributes/translate) HTML global attribute.

## Value

> A boolean value that is `true` if an element's attribute values and the values of its Text node children are to be translated when the page is localized, `false` otherwise.

Reviewer: ```suggestion
A boolean value that is `true` if an element's attribute values and the values of its {{domxref("Text")}} node children are to be translated when the page is localized, `false` otherwise.
```

## Examples

> The following example shows how to enable or disable translation when localize via script:

Reviewer: ```suggestion
The following example shows how to enable or disable translation via script:
```

> ```js

Reviewer: Could we try to make this a live example?

const translateElement = document.querySelector(".translate-element");
const noTranslateElement = document.querySelector(".no-translate-element");

> // enable translation when localize

Reviewer: ```suggestion
// Enable translation
```

> if (!translateElement.translate) {
>   translateElement.translate = true;
> }

Reviewer: The test is useless.

If `translateElement.translate` is `true`, it stays `true`.
if `translateElement.translate` is `false`, it switches to `true`.

```suggestion
translateElement.translate = true;
```

> // disable translation when localize

Reviewer: ```suggestion
// Disable translation
```

> if (noTranslateElement.translate) {
>   noTranslateElement.translate = false;
> }

Reviewer: ```suggestion
noTranslateElement.translate = false;

```

---
title: "HTMLElement: translate property"
short-title: translate
slug: Web/API/HTMLElement/translate
page-type: web-api-instance-property
browser-compat: api.HTMLElement.translate
---

{{APIRef("HTML DOM")}}

The **`translate`** property of the {{domxref("HTMLElement")}} interface indicates whether an element's attribute values and the values of its {{domxref("Text")}} node children are to be translated when the page is localized, or whether to leave them unchanged.

It reflects the value of the [`translate`](/en-US/docs/Web/HTML/Global_attributes/translate) HTML global attribute.

## Value

A boolean value that is `true` if an element's attribute values and the values of its {{domxref("Text")}} node children are to be translated when the page is localized, `false` otherwise.

## Examples

The following example shows how to enable or disable translation via script:

```html

> <div>
>   <span>The content will always be translated:</span>
> </div>
> <div>
>   <span translate="yes">El contenido será traducido</div>
> </div>
> <div>
>   <span id="translate-label">The content will be translated:</span>
> </div>
> <div>
>   <span id="translate-element" translate="no">El contenido puede ser traducido.</div>
> </div>

Reviewer: ```suggestion
<div>
  <span>The content may always be translated: </span>
  <span translate="yes">El contenido será traducido</div>
</div>
<div>
  <span id="translate-label">The content may be translated:</span>
  <span id="translate-element" translate="no">El contenido puede ser traducido.</div>
</div>
```

> <div>
>   <span>Change the checkbox to enable or disable translation</span>
> </div>
> <div>
>   <input id="translate-controller" type="checkbox" />
> </div>

Reviewer: ```suggestion
<input id="translate-controller" type="checkbox" /> Enable translation
```

```

```js
const label = document.getElementById("translate-label");
const element = document.getElementById("translate-element");
const controller = document.getElementById("translate-controller");

> controller.addEventListener("change", (e) => {
>   if (controller.checked) {
>     element.translate = true;
>     label.innerText = "The content will be translated:";
>   } else {
>     element.translate = false;
>     label.innerText = "The content will not be translated:";
>   }
> });

Reviewer: ```suggestion
controller.addEventListener("change", (e) => {
  if (controller.checked) {
    element.translate = true;
    label.innerText = "The content may be translated:";
  } else {
    element.translate = false;
    label.innerText = "The content may not be translated:";
  }
});
```
